index.d.ts

```diff
@@ -81,6 +81,9 @@ declare module '@primer/components' {
     children?: DetailsRenderFunction | React.ReactNode
     defaultOpen?: boolean
     overlay?: boolean
+    open?: boolean
+    onToggle?: (event: React.SyntheticEvent<HTMLDetailsElement>) => void
+    onClickOutside?: React.MouseEventHandler
```

Comment:
I think maybe you want this to be 

`onClickOutside?: (event: MouseEvent) => void `

It looks like `React.MouseEventHandler` uses a `React.MouseEvent` but in this case I think you need the native `MouseEvent` instead?

In this code sandbox, I'm able to use `listener2` in my call to `addEventListener` but not `listener1`.
https://codesandbox.io/s/solitary-dust-fil2m